mm2src/lp_ordermatch.rs

```diff
@@ -3978,8 +3981,7 @@ pub async fn update_maker_order(ctx: MmArc, req: Json) -> Result<Response<Vec<u8
 
             let new_change = HistoricalOrder::build(&update_msg, order);
             order.apply_updated(&update_msg);
-            order.changes_history.get_or_insert(Vec::new()).push(new_change);
-            save_maker_order_on_update(ctx.clone(), order).await;
+            save_maker_order_on_update(ctx.clone(), order, new_change).await;
```

Comment:
Can the following actually replace all the code of this PR?
```rust
if order.save_in_history {
    order.changes_history.get_or_insert(Vec::new()).push(new_change);
}
```
Or maybe it's worth adding `store_update_history` field to the order with a following logic:
```rust
if order.store_update_history {
    order.changes_history.get_or_insert(Vec::new()).push(new_change);
} else {
    order.changes_history = None
}
```

Reply:
the problem with these 2 suggestions is that if a user wants to use save_in_history/store_update_history, it will lead to memory leaks too. The current solution moves the cause of the memory leak to storage so there would be non-memory leaks for all cases.

Reply:
I see, it makes sense, thanks.